Week3/script.js

```diff
@@ -0,0 +1,273 @@
+/* *** String! *** */
+
+// 1.1
+
+let myString = "hello,this,is,a,difficult,to,read,sentence";
+
+console.log(myString);
+
+// 1.2
+
+console.log(myString.length);
+
+// 1.3 & 1.4
+
+console.log(myString.replace(/,/g, " "));
+
+/* ***Arrays!*** */
+
+let favoriteAnimals = ["blowfish", "capricorn", "giraffe"];
+
+// 2.1
+
+favoriteAnimals.push("turtle");
+
+// 2.2
+
+console.log(favoriteAnimals);
+
+// 2.3
+
+favoriteAnimals.splice(1, 0, "meerkat");
+
+// 2.4
+
+console.log(
+  'favoriteAnimals will be: ["blowfish", "meerkat", "capricorn", "giraffe", "turtle"] '
+);
+
+// 2.5
+
+console.log(favoriteAnimals);
+
+// 2.6
+
+console.log("The array has a length of : 5");
+
+// 2.7
+
+favoriteAnimals.splice(3, 1);
+
+// 2.8
+
+console.log(favoriteAnimals);
+
+// 2.9
+
+console.log(favoriteAnimals.indexOf("meerkat"));
+
+// 2.10
+
+console.log("The item you are looking for is at index: 1");
+
+        /* *** More JavaScript *** */
+
+        // 1
+
+function threeArguments(a, b, c) {
+  var sum = a + b + c;
+
+  return sum;
+}
+
+        // 2
+
+function colorCar(car) {
+  return car;
+}
+
+let car = "a red car";
+
+console.log(colorCar(car));
+
+        // 3
+
+const person = {
+  name: "Mehmet Ekinci",
+  gender: "M",
+  adres() {
+    console.log("Koemeershoeve");
+  }
+};
+console.log(person);
+
+        // 4
+
+function vehicleType(color,code) {
+  let type = "";
+  if (code === 1) {
+    type = " car";
+  } if (code === 2) {
+    type = " motorbike";
+  }
+
+  console.log("a "+ color + type)
+
+}
+
+vehicleType("blue", 2);
+
+        // 5
+
+console.log(3 === 3 ? "yes" : "no")
+
+        // 6
+
+
+
+function vehicle(color,code,age) {
+  let type,condition = "";
+
+  if (code === 1) {
+    type = " car";
+  } if (code === 2) {
+    type = " motorbike";
+  } if (age >= 2) {
+    condition = " used";
+  } if ((age >= 0) && (age <= 1)) {
+    condition = " new";
+  }
+  
+  console.log("a " + color + condition + type);
+
+}
+vehicle("blue",1,5);
+
+        // 7
+
+let vehicles = ["motorbike", "caravan", "bike", "car", ]
+
+        // 8
+
+console.log(vehicles[2]);
+
+        // 9
+
+function newVehicle(color,code,age) {
+  let type,condition = "";
+  if (code === 1) {
+    type = " motorbike";
+  } if (code === 2) {
+    type = " caravan";
+  }  if (code === 3) {
+    type = " bike";
+  } if (code === 4) {
+    type = " car";
+  } if (age >=2) {
+    condition = " used";
+  } if ((age >= 0) && (age <=1)) {
+    condition = " new";
+  }
+  
+  console.log("a " + color + condition + type)
+}
+newVehicle("green",3,1);
+
+        // 10
+let advertisement = "Amazing Joe's Garage, we service "
+function reAdvertisement(advertisement, vehicles){
+
+  for( let i = 0; i < vehicles.length ;i++){
+    const lastElement = vehicles[vehicles.length-1];
+    const beforLastElement = vehicles[vehicles.length-2];
+    const currentElement = vehicles[i];
+
+    if (currentElement === lastElement) {
```

Comment:
Ah nice!

This looks good, you can also use the `indexes` though and then compare the `index` numbers. So then it would look like this:

```
for( let i = 0; i < vehicles.length ;i++){
    if (i=== vehicles.length - 1) {
     advertisement = advertisement + currentElement + ".";
    } else if (i=== vehicles.length - 2) {
      advertisement = advertisement + currentElement + " and ";
    } else {
      advertisement = advertisement + currentElement + ", "
    }
```

Reply:
Ah, both ways are fine:) I think the one I suggested is a bit easier (and less error-prone with the comparisons). So it's a `safer` option to compare numbers instead of whole objects:)